src/sharding.js

```diff
@@ -56,14 +77,22 @@ class ShardingDatastore extends Adapter {
     return ShardingDatastore.open(store)
   }
 
+  /**
+   * @param {Datastore} store
+   */
   static async open (store) {
     const shard = await sh.readShardFun('/', store)
     return new ShardingDatastore(store, shard)
   }
 
+  /**
+   * @param {Datastore} store
+   * @param {Shard} shard
+   */
   static async create (store, shard) {
     const exists = await store.has(shardKey)
     if (!exists) {
+      // @ts-ignore i have no idea what putRaw is or saw any implementation
```

Comment:
@achingbrain do you know ?

Reply:
datastore-fs has putRaw to write and read without extension